### Regresión Lineal Múltiple - High Point Sofa and Chair
1. Determinar las variables independientes
2. Determinar la variable dependiente
3. Obtener el valor de interseccion
4. Estimar el corto de calefaccion para una Temperatura externa media de 30 grados, 5 pulgadas de aislamiento y un calentador con antiguedad de 10 años.

In [3]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [4]:
# Importar datos
df = pd.read_csv('data/HighPointSofaandChair.csv')
df.head()

Worker  Performance  Aptitude  Union
0    Abbott           58         5      0
1  Anderson           53         4      0
2    Bender           33        10      0
3      Bush           97        10      0
4    Center           36         2      0

# Determinar las variables independientes
x1,x2,y3

In [5]:
X = df[['Aptitude', 'Union']]
X

Aptitude  Union
0          5      0
1          4      0
2         10      0
3         10      0
4          2      0
5          7      0
6          6      0
7          9      0
8          9      1
9          2      1
10         8      1
11         6      1
12         7      1
13         3      1
14         6      1

In [6]:
# Determinar las variables Dependientes
y = df[['Performance']]
y

Performance
0            58
1            53
2            33
3            97
4            36
5            83
6            67
7            84
8            98
9            45
10           97
11           90
12           96
13           66
14           82

In [7]:
# Ajustar el modelo de regresión multiple
model = LinearRegression()
model.fit(X,y) # entrenando el modelo
y_pred = model.predict(X)
#y_pred

# Métricas estadísticas
mse = mean_squared_error(y, y_pred) # Error de la media cuadrática
r2 = r2_score(y,y_pred) # varianza

# Calcular el valor de Interseccion
Cuando la aptitude y union son 0

In [8]:
model.intercept_

array([29.27618014])

# Coeficiente
En cuanto aumenta el performance por cada punto de aptitude y union

# Error cuadrático medio
Mientras más bajo mejor, el modelo se equivoca por 15.13 puntos en promedio

# Varianza 
El modelo explica aproximadamente el 53.3% de la variación del desempeño laboral (0% el modelo no explica nada)

In [9]:
print("Coeficientes: ", model.coef_)
print("Error cuadrático medio",mse)
print("Varianza: ", r2)

Coeficientes:  [[ 5.22246337 22.13510581]]
Error cuadrático medio 228.9361548200398
Varianza:  0.5329951510923939


In [10]:
Y = model.predict([[6,0]])
print("El costo de calefaccion estimado es: ", Y[0])

El costo de calefaccion estimado es:  [60.61096039]


c:\Users\herna\anaconda3\envs\notebooks\Lib\site-packages\sklearn\utils\validation.py:2749: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Formula
Performance = 29.2761 + 5.222(Aptitude) + 22.135(Union)

# Como se soluciono el problema con la regresión lineal multiple?
Se pudo ver que con la regresión lineal multiple 

# Cual es la relacion entre los datos obtenido?
Los resultados de la regresión muestran que existe una relación positiva entre las variables independientes y el desempeño laboral: por cada punto adicional en la calificación del examen de aptitud (Aptitude), el desempeño laboral (Performance) aumenta en promedio aproximadamente 5.22 puntos, manteniendo constante la pertenencia al sindicato. Además, pertenecer al sindicato (Union = 1) incrementa el desempeño estimado en alrededor de 22.13 puntos en comparación con un trabajador no sindicalizado (Union = 0), lo que indica que ambas variables influyen significativamente en el desempeño laboral.